### ISSUE TITLE
Service `dependencies` option

### ISSUE THREAD

**Issue by tinchoz49** [ice-services/moleculer#102]
Hi @icebob, as we talk, here is the repository:

https://github.com/tinchoz49/example-moleculer-wait-for-services

**Comment by icebob** [comment]
Thank you!

**Comment by icebob** [comment]
@tinchoz49 I checked it. If you `return` with the `waitForServices` in `started`, it will cause a dead lock, because transporter wait for all local services started, but the service wait for receive other services via transporter. Workaround is to remove the `return` in `started`. But I will think out something to solve it.

**Comment by tinchoz49** [comment]
maybe we can define a hook method `started` or like an option `waitForServices` in the moleculer.config.js to hook up in the `broker.start()` promise.

Something like this:
```javascript
// moleculer.config.js
module.exports = {
  namespace: 'dev',
  nodeID: 'node-' + process.pid,

  waitForServices: ['greeter-two'],
  ...
};
```

**Comment by WoLfulus** [comment]
@tinchoz49 I'd sugest something like `dependencies` or `depends_on` instead of `waitForServices` in this case

**Comment by icebob** [comment]
I vote to `dependencies` but it would be better to place in the service schema because every schema has different dependencies.
Example: 
```js
// posts.service.js
module.exports = {
  name: "posts",
  dependencies: ["users", "comments"]
  ....
}
```
In this case the broker will call the "posts" service `started` handler if all dependent services are available.
What do you think?

**Comment by tinchoz49** [comment]
I like the suggestion! my vote for `dependencies` at the service schema level.

**Comment by WoLfulus** [comment]
It doesn't make sense to fire `started` if dependencies are still pending, so yeah, that works. :)

**Comment by icebob** [comment]
After #112 it should support versioned services too.
Syntax:
```js
// posts.service.js
module.exports = {
  name: "posts",
  dependencies: [
      { name: "users", version: 2 },
      { name: "comments", version: "staging" }
  ]
  ....
}
```
And need an internal settings for timeout:
```js
module.exports = {
  name: "posts",
  settings: {
      $dependencyTimeout: 30000
  }
}
```

**Comment by icebob** [comment]
Done. Will be released in v0.11.2 (this week)

**Comment by tinchoz49** [comment]
thanks for making each day this library more awesome!

**Comment by icebob** [comment]
Released in v0.11.2
